### DISCUSSION TITLE
Correct way to Set/Get Character Mass

### DISCUSSION THREAD

**Discussion by ktimam** [jrouwe/JoltPhysics#869] [Q&A] [answered]
What is the correct way to Set then get the mass for a character? 
I tried the following, but the returned mass is different:

  // Create 'player' character
  Ref<CharacterSettings> settings = new CharacterSettings();
  settings->mMass = 3;
  Character* character;
character = new Character(settings, Vec3(200,0,100), Quat::sIdentity(), 0, GetPhysicsSystem());
character->AddToPhysicsSystem(EActivation::Activate); 
float mass = GetBodyInterface().GetShape(character->GetBodyID()).GetPtr()->GetMassProperties().mMass;

**Comment by jrouwe** [comment, marked as answer]
```Shape::GetMassProperties``` returns the mass as calculated from the density of the shape, this is the default mass if you don't override it.

You want something like:

```
BodyLockRead lock(GetPhysicsSystem().GetBodyLockInterface(), character->GetBodyID());
if (lock.Succeeded())
{
	float mass = 1.0f / lock.GetBody().GetMotionProperties()->GetInverseMass();
}
```
